pconnector/client/option_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"os/exec"
+	"slices"
+	"testing"
+
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+	"github.com/hashicorp/go-plugin"
+	"github.com/matryer/is"
+)
+
+func TestOption_WithConnectorUtilsAddress(t *testing.T) {
+	is := is.New(t)
+	cmd := exec.Command("test-path")
+	cmd.Env = []string{"FOO=BAR"}
+	cc := &plugin.ClientConfig{Cmd: cmd}
+
+	underTest := WithEnvVar(pconduit.EnvConduitConnectorUtilitiesGRPCTarget, "localhost:12345")
+
+	err := underTest.ApplyOption(cc)
+	is.NoErr(err)
+
+	want := []string{
+		"FOO=BAR",
+		pconduit.EnvConduitConnectorUtilitiesGRPCTarget + "=localhost:12345",
+	}
+	slices.Sort(want)
+	slices.Sort(cc.Cmd.Env)
```

Comment:
Nitpick - I don't think we should sort the outcome in this test. The order of env vars plays a role in case they are repeated:

```sh
$ A=foo B=bar; echo "$A $B"
foo bar
$ A=foo B=bar A=baz; echo "$A $B"
baz bar
```